jquery.simulate.js

```diff
@@ -200,8 +200,8 @@ $.extend( $.simulate.prototype, {
 		return event;
 	},
 
-	dispatchEvent: function( elem, type, event ) {
-		if ( elem[ type ] ) {
+	dispatchEvent: function( elem, type, event, options ) {
+		if ( elem[ type ] && !options ) {
```

Comment:
> Generally though, don't we want to use dispatchEvent() by default and use click() as the fallback? In other words, should we change the order of those tests?

Yes, I think that is right but just haven't done the research to be sure. So something like this, along with some more extensive unit tests:
```js
	dispatchEvent: function( elem, type, event, options ) {

		// Standard W3C event dispatch
		if ( elem.dispatchEvent ) {
			elem.dispatchEvent( event );

		// Ensure the checkbox/radio state changes on IE<=8 (Issue #1)
		} else if ( type === "click" && elem.click && elem.nodeName.toLowerCase() === "input" )
			elem.click(); 

		// IE<=8 event dispatch
		} else if ( elem.fireEvent ) {
			elem.fireEvent( "on" + type, event );
		}
	},
```